src/app/shared/components/breadcrumb.tsx

```diff
@@ -3,165 +3,127 @@
-                const moduleId = this.getModuleIdFromSearch(search);
-                items.push({
-                    href: ``,
-                    key: `device_${deviceId}_${moduleId}`,
-                    text: moduleId
-                });
-                return;
-            }
-        });
+    private getShortHubName = (hostName: string) => {
+        return hostName && hostName.replace(/\..*/, '');
+    }
 
-        return items;
+    private getLocalizedKey = (t: TranslationFunction, route: string | undefined) => {
+        const resourceKey = (ResourceKeys.breadcrumb as any)[route]; // tslint:disable-line:no-any
+        return resourceKey ? t(resourceKey) : route;
     }
 
-    private buildDeviceDetails = (pathComponents: string[], hubName: string, deviceId: string, t: TranslationFunction): BreadcrumbEntry[] => {
-        const entries: BreadcrumbEntry[] = [];
-        const paths = [...pathComponents];
-        const query = `${ROUTE_PARAMS.DEVICE_ID}=${deviceId}`;
-        let basePath = `#/${ROUTE_PARTS.RESOURCE}/${hubName}/${ROUTE_PARTS.DEVICES}/${ROUTE_PARTS.DEVICE_DETAIL}`;
-        while (paths.length > 0) {
-            basePath = `${basePath}/${paths[0]}`;
-            if (t((ResourceKeys.deviceContent.navBar as any)[paths[0]])) { // tslint:disable-line:no-any
-                entries.push({
-                    href: paths.length !== 1 ? `${basePath}/?${query}` : '',
-                    key: paths[0],
-                    text: t((ResourceKeys.deviceContent.navBar as any)[paths[0]]) // tslint:disable-line:no-any
-                });
-            }
-            paths.splice(0, 1);
+    private getCurrentUrl = () => {
+        let url = this.props.match.url;
+        if (url.endsWith('/')) {
+            url = url.slice(0, -1); // remove trailing slash
         }
-
-        return entries;
+        return url.split('/').pop();
     }
 
-    private buildDigitalTwinDetails = (pathComponents: string[], hubName: string, deviceId: string, interfaceId: string, t: TranslationFunction): BreadcrumbEntry[] => {
-        const entries: BreadcrumbEntry[] = [];
-        const paths = [...pathComponents];
-        const query = `${ROUTE_PARAMS.DEVICE_ID}=${deviceId}&${ROUTE_PARAMS.INTERFACE_ID}=${interfaceId}`;
-        let basePath = `#/${ROUTE_PARTS.RESOURCE}/${hubName}/${ROUTE_PARTS.DEVICES}/${ROUTE_PARTS.DEVICE_DETAIL}/${ROUTE_PARTS.DIGITAL_TWINS}`;
-        while (paths.length > 0) {
-            basePath = `${basePath}/${paths[0]}`;
-            entries.push({
-                href: paths.length !== 1 ? `${basePath}?${query}` : '',
-                key: paths[0],
-                text: t((ResourceKeys.deviceContent.navBar as any)[paths[0]]) // tslint:disable-line:no-any
+    private renderLinkItem = (linkContent: string, searchParams?: ROUTE_PARAMS[]) => {
+        const { match } = this.props;
+        let href = match.url;
+        if (searchParams) {
+            href += '/';
```

Comment:
you should remove this otherwise default case in the switch will also have an extra '/' at the end.

Reply:
default case has no trailing slash, only when search parameters are present